catalyst_voices/apps/voices/lib/widgets/document_builder/document_checkbox_builder_widget.dart

```diff
@@ -0,0 +1,123 @@
+  });
+
+  @override
+  State<DocumentCheckboxBuilderWidget> createState() =>
+      _DocumentCheckboxBuilderWidgetState();
+}
+
+class _DocumentCheckboxBuilderWidgetState
+    extends State<DocumentCheckboxBuilderWidget> {
+  bool editMode = false;
+  late bool value;
+  late bool initialValue;
+  DocumentNodeId get nodeId => widget.property.schema.nodeId;
+  String get description => widget.property.schema.description ?? '';
+  AgreementConfirmationDefinition get definition =>
+      widget.property.schema.definition as AgreementConfirmationDefinition;
+
+  @override
+  void initState() {
+    super.initState();
+
+    value = bool.tryParse(widget.property.value.toString()) ??
+        bool.tryParse(widget.property.schema.defaultValue.toString()) ??
+        false;
+    initialValue = value;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return DecoratedBox(
+      decoration: BoxDecoration(
+        color: Theme.of(context).colors.elevationsOnSurfaceNeutralLv1White,
+      ),
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 24, vertical: 12),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            DocumentPropertyTopbar(
+              isEditMode: editMode,
+              onToggleEditMode: _toggleEditMode,
+              title: widget.property.schema.title,
+            ),
+            const SizedBox(height: 22),
+            Offstage(
+              offstage: description.isEmpty,
+              child: Text(
+                description,
+                style: Theme.of(context).textTheme.titleSmall,
+              ),
+            ),
+            const SizedBox(height: 22),
+            VoicesCheckbox(
+              value: value,
+              onChanged: _changeValue,
+              isDisabled: !editMode,
+              label: Text(
+                context.l10n.agree,
+                style: Theme.of(context).textTheme.bodyLarge?.copyWith(
+                      color: !editMode
+                          ? Theme.of(context).colors.textDisabled
+                          : null,
+                    ),
+              ),
+            ),
+            Offstage(
+              offstage: !editMode,
+              child: DocumentPropertyFooter(
+                onSave: !_isValidValue
+                    ? null
+                    : () {
+                        widget.onChanged(
+                          DocumentChange(
+                            nodeId: nodeId,
+                            value: value,
+                          ),
+                        );
+                        setState(() {
+                          editMode = false;
+                          initialValue = value;
+                        });
+                      },
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void _toggleEditMode() {
+    if (editMode) {
+      value = initialValue;
+    }
```

Comment:
I think names of those properties are confusing.

Now `initialValue` is a `value` (updated after each save) and `value` is whatever user sees in edit mode even. Think of more descriptive names please